shim/src/ctrl/getsockopt.c

```diff
@@ -35,12 +36,35 @@ int __getsockopt(int sockfd, int level, int optname, void *optval, socklen_t *op
 
     TRACE("sockfd=%d, level=%d, optname=%d, optval=%p, optlen=%p", sockfd, level, optname, optval, (void *)optlen);
 
-    // TODO: Hook in demi_getsockopt().
-    UNUSED(level);
-    UNUSED(optname);
-    UNUSED(optval);
-    UNUSED(optlen);
-    UNIMPLEMETED("demi_getsockopt() is not hooked in");
+    // Issue warnings for common options that are not supported.
+    if (level == SOL_SOCKET && optname == SO_KEEPALIVE)
+    {
+        WARN("%s is not supported", "SO_KEEPALIVE");
+    }
+    else if (level == SOL_SOCKET && optname == SO_REUSEADDR)
+    {
+        WARN("%s is not supported", "SO_REUSEADDR");
+    }
+    else if (level == IPPROTO_TCP && optname == TCP_NODELAY)
```

Comment:
Is there a reason that TCP_NODELAY is not supported? We support it

Reply:
It is not supported by `demi_getsockopt()`:

https://github.com/microsoft/demikernel/blob/4b1ed79c2216d1e8c908292129193f9331fec517/src/rust/demikernel/bindings.rs#L863-L869

Reply:
Oh, that is in my recent PR! Ha, you might want to wait to merge this until PR #1274 is merged.